server/liveblog/themes/themes.py

```diff
@@ -271,17 +271,24 @@ def publish_related_blogs(self, theme):
         # for t in self.get_children(theme['name']) + [theme['name']]:
         #     terms.append({'term': {'blog_preferences.theme': t}})
         blogs = get_resource_service('blogs').get(req=None, lookup={})
+        outputs = get_resource_service('outputs').get(req=None, lookup={})
         # get all the children for the theme that we modify the settings for
         theme_children = self.get_children(theme.get('name'))
         for blog in blogs:
             blog_pref = blog.get('blog_preferences')
-            if blog_pref['theme'] == theme['name']:
+            if blog_pref.get('theme') == theme['name']:
```

Comment:
When dealing with each individual blog which uses the theme which has been changed, you should then get outputs for this blog which use the same theme (i.e. which have a null theme)